### DISCUSSION TITLE
Click event works neither on PC nor on Mobile

### DISCUSSION THREAD

**Discussion by gabrielefiorenza** [AR-js-org/AR.js#337] [Help]
I cannot add a click event on an a-image on my AR.js html built for mobile.

Here is my code, but it works neither on pc nor on mobile.

Can someone tell me why?

```
`<script src="https://aframe.io/releases/1.0.4/aframe.min.js"></script>
<script src="https://raw.githack.com/AR-js-org/AR.js/master/aframe/build/aframe-ar.js"></script>
<script>        
    AFRAME.registerComponent('clicker', {
        init: function() {
            this.el.addEventListener('click', e => {
                alert('Clicked!');
            });
        }
    });
</script>

<body style="margin : 0px; overflow: hidden;">
    <a-scene embedded arjs="sourceType: webcam; debugUIEnabled: false; detectionMode: mono_and_matrix; matrixCodeType: 3x3;" cursor="fuse: false; rayOrigin: mouse">

        <a-assets>
            <img id="linkedin" src="linkedIn logo.png">
        </a-assets>

        <a-marker type="barcode" value="6">
            <a-image clicker id="element_linkedin" src="#linkedin"  rotation="280 0 0" scale="0.001 0.001 0.001" position="1.5 0 1" animation="property: scale; to: 0.7 0.7 0.7; dur: 2500; easing: linear; loop: false; startEvents: animation_linkedin"></a-image>
        </a-marker>

        <a-entity camera></a-entity>
    </a-scene>
</body>`
```

**Comment by nickw1** [comment]
Hi, that example I think is outdated and will not work on recent versions of A-Frame.

Have a look at the source of:

https://hikar.org/webapp/ex1.html

This one works on my devices.

Note the raycaster component on the `a-scene`, and its near and far properties. These define the start and end of a ray from the mouse click position into the screen: any object along this ray will receive a click event. In other words, in this example, if the object is up to 10000 units away it will be detected.

**Reply by gabrielefiorenza** [reply]
Hi, thank you for your reply.

I have tried to use the code of the example you gave me by adapting it to my situation.

It seems to work but:

1) only with a-box not with a.-image, which is what I need

2) it seems that the spot for firing the click event is not right on the box but it is a little shifted on the left.

Can you tell me why?

Thank you so much!

**Reply by nickw1** [reply]
Hi, I have to admit I am not sure of this; I have only tested it on primitives like boxes and other geometries.

I have the feeling this issue has come up before; all I can suggest at the moment is to look at the issues and see if anything has been suggested.

Sorry I can't be more helpful!
